docs/cow-protocol/tutorials/arbitrate/solver-engine.md

```diff
@@ -1 +1,98 @@
+---
+sidebar_position: 1
+draft: true
+---
 
+# Solver Engine
+
+The Solver Engine implements the [matching system](https://en.wikipedia.org/wiki/Order_matching_system) of the CoW Protocol batch auction.
+As part of the auction lifecycle, it receives a request from the [driver](driver) containing a potentially pre-processed auction instance and returns information about which order to execute together with instructions on how to achieve the desired execution using on-chain liquidity "interactions". 
+
+:::note
+
+Solver teams aiming to use the open-source driver implementation, need to implement their own solver engine.
+
+:::
+
+## Overview
+
+An open source [Rust](https://rust-lang.org)implementation of a solver engine can be found in the [solvers crate](https://github.com/cowprotocol/services/tree/main/crates/solvers).
+This engine can be run in a few different "modes" to demonstrate and test the whole system end to end (cf. `solvers::infra::cli::Command`):
+
+- **Baseline**: Solve individual orders exclusively via on-chain liquidity provided in the driver-augmented auction.
+- **Naive**: Optimistically batch similar orders and settle net-amounts via AMMs
+- **Legacy**: Forward auction to another solver-engine implementing the deprecated, legacy HTTP interface
```

Comment:
```suggestion
- **Legacy**: Forward auction to another solver engine implementing the deprecated, legacy HTTP interface
```

(Otherwise I read "solver-engine implementing" as a single compound adjective.)